mautic/mautic
# Import configuration enhancements

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | No
| New feature? | Yes
| Automated tests included? | No
| Related user documentation PR URL | In-line (Configuration page)
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | None
| Deprecations? | None

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description: This PR makes the following enhancements to Imports
1. allows the imports directory to be configured. 
2. limits the notifications from the import command, only sending a notification when processing is done (i.e., STOPPED, FAILED, COMPLETE)
3. allows the timeout and subsequent status to be configured
a. DELAYED to attempt a restart automatically
b. STOPPED to allow a manual retry
c. FAILED to cancel the job and remove the file
4. It also fixes a few smaller issues encountered along the way 

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Open Configuration page
3. Modify the setting in the imports Dir input.and save the changes
4. Go to Contacts->Import and create a new import for background processing
5. Verify that the import file was copied to the new import directory
